feat : use "PIP_NO_CACHE_DIR" env with pip in dockerfiles to save space

rajpratik71 opened the pull request:
using the  "PIP_NO_CACHE_DIR" env with pip install, make sure downloaded packages by pip don't cache on the system. This is a best practice that makes sure to fetch from a repo instead of using a local cached one. Further, in the case of Docker Containers, by restricting caching, we can reduce image size. In terms of stats, it depends upon the number of python packages multiplied by their respective size. e.g for heavy packages with a lot of dependencies it reduces a lot by don't cache pip packages.

Further, more detailed information can be found at

https://medium.com/sciforce/strategies-of-docker-images-optimization-2ca9cc5719b6

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

intrigus-lgtm commented:
Instead of changing the files, can't we simply set `ENV PIP_NO_CACHE_DIR=1` in the Dockerfile(s)?

disconnect3d commented:
> Instead of changing the files, can't we simply set `ENV PIP_NO_CACHE_DIR=1` in the Dockerfile(s)?

+1, this sounds like a better solution (?)

disconnect3d reviewed (changes requested):
Please change this to envvar in dockerfiles instead of flag to pip in setup.sh/setup-dev.sh

disconnect3d commented:
Ping @Rajpratik71

rajpratik71 commented:
Sorry @intrigus-lgtm  @disconnect3d  for the late response.

Env variable and value related to "PIP_NO_CACHE_DIR" are bit confusing i.e. why I thought to keep flag-based, see the thread at https://github.com/pypa/pip/issues/2897

intrigus-lgtm commented:
This is no longer a problem.
I don't have time to find the relevant issue that fixes this, but it should be no problem with any recent pip/python version.
EDIT: https://github.com/pypa/pip/pull/5884

rajpratik71 committed:
feat : use "PIP_NO_CACHE_DIR" env with pip in dockerfiles to save space

using the  "PIP_NO_CACHE_DIR" env with pip install, make sure downloaded packages by pip don't cache on the system. This is a best practice that makes sure to fetch from a repo instead of using a local cached one. Further, in the case of Docker Containers, by restricting caching, we can reduce image size. In terms of stats, it depends upon the number of python packages multiplied by their respective size. e.g for heavy packages with a lot of dependencies it reduces a lot by don't cache pip packages.

Further, more detailed information can be found at

https://medium.com/sciforce/strategies-of-docker-images-optimization-2ca9cc5719b6

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

disconnect3d commented:
> Sorry @intrigus-lgtm @disconnect3d for the late response.
> 
> Env variable and value related to "PIP_NO_CACHE_DIR" are bit confusing i.e. why I thought to keep flag-based, see the thread at [pypa/pip#2897](https://github.com/pypa/pip/issues/2897)

We wanted it to be added to Dockerfiles, not to setup.sh itself :P

This still needs changes :<

disconnect3d commented:
Please either reopen this or send another PR with proper fix for this.

rajpratik71 commented:
@disconnect3d  Sorry for the confusion and delayed response new PR is ta https://github.com/pwndbg/pwndbg/pull/2068
